nutanix/resource_nutanix_ndb_register_database.go

```diff
@@ -0,0 +1,697 @@
+		}
+
+		if err := d.Set("database_status", resp.Databasestatus); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("dbserver_logical_cluster_id", resp.Dbserverlogicalclusterid); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("time_machine_id", resp.Timemachineid); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("parent_time_machine_id", resp.Parenttimemachineid); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("time_zone", resp.Timezone); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("info", flattenDBInfo(resp.Info)); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("group_info", resp.GroupInfo); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("metadata", flattenDBInstanceMetadata(resp.Metadata)); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("metric", resp.Metric); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("category", resp.Category); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("parent_database_id", resp.ParentDatabaseID); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("parent_source_database_id", resp.ParentSourceDatabaseID); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("lcm_config", flattenDBLcmConfig(resp.Lcmconfig)); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("time_machine", flattenDBTimeMachine(resp.TimeMachine)); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("dbserver_logical_cluster", resp.Dbserverlogicalcluster); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("database_nodes", flattenDBNodes(resp.Databasenodes)); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("linked_databases", flattenDBLinkedDbs(resp.Linkeddatabases)); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	return nil
+}
+func resourceNutanixNDBRegisterDatabaseUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*Client).Era
+	if c == nil {
+		return diag.Errorf("era is nil")
+	}
+
+	dbID := d.Id()
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+
+	updateReq := era.UpdateDatabaseRequest{
+		Name:             name,
+		Description:      description,
+		Tags:             []interface{}{},
+		Resetname:        true,
+		Resetdescription: true,
+		Resettags:        true,
+	}
+
+	res, err := c.Service.UpdateDatabase(ctx, &updateReq, dbID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if res != nil {
+		if err = d.Set("description", res.Description); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err = d.Set("name", res.Name); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+	return nil
+}
+
+func resourceNutanixNDBRegisterDatabaseDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
Isn't it same as database resource delete ?

Reply:
yes. but with different inputs.